config/recipes/beats/filebeat_autodiscover_by_metadata.yaml

```diff
@@ -0,0 +1,115 @@
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: filebeat
+spec:
+  type: filebeat
+  version: 7.8.0
+  elasticsearchRef:
+    name: elasticsearch
+  kibanaRef:
+    name: kibana
+  config:
+    filebeat.autodiscover.providers:
+    - node: ${HOSTNAME}
+      type: kubernetes
+      hints.default_config.enabled: "false"
+      templates:
+      - condition.equals.kubernetes.namespace: log-namespace
+        config:
+        - paths: ["/var/log/containers/*${data.kubernetes.container.id}.log"]
+          type: container
+      - condition.equals.kubernetes.labels.log-label: "true"
+        config:
+        - paths: ["/var/log/containers/*${data.kubernetes.container.id}.log"]
+          type: container
+    processors:
+    - add_cloud_metadata: {}
+    - add_host_metadata: {}
+  daemonSet:
+    podTemplate:
+      spec:
+        serviceAccountName: filebeat
+        automountServiceAccountToken: true
+        terminationGracePeriodSeconds: 30
+        dnsPolicy: ClusterFirstWithHostNet
```

Comment:
I realize this is in the Beats-provided examples, but it's not clear to me why filebeat would need access to the host network namespace. I didn't get a chance to test it without it yet though

Reply:
I checked with @exekias and while it works without it, host metadata will be richer - `host.ip`, `host.mac` will be present. I'd be fine leaving it as it is.

Reply:
I think we at least want to leave a comment, I can't be the only one who sees this and it rings alarm bells. I'm also skeptical of how useful the host network info is for container logs.